getting-started/basicapp/README.md

```diff
@@ -0,0 +1,362 @@
+example on Mac, use the commands
+
+```
+curl -o cloud_sql_proxy https://dl.google.com/cloudsql/cloud_sql_proxy.darwin.386
+chmod +x cloud_sql_proxy
+```
+
+Start the proxy with the commands
+
+```
+PROJECT=[your project]
+REGION=[your region]
+INSTANCE_NAME=basicapp
+CONNECTION_NAME=$PROJECT:$REGION:$INSTANCE_NAME
+./cloud_sql_proxy -instances="$CONNECTION_NAME"=tcp:3306
+```
+
+Run the web app in a separate terminal
+```
+DBPASSWORD=[your password]
+DBUSER=proxyuser
+DBHOST=127.0.0.1
+DATABASE=messagesdb
+export MYSQL_CONNECTION: "$DBUSER:$DBPASSWORD@tcp($DBHOST:3306)/$DATABASE"
+go run cmd/basicwebapp/basicwebapp.go
+```
+
+You can now use the same curl commands as above to send your friend a message.
+
+## Packaging in a Docker Container
+The basicapp example can package your application in a Docker container based
+on the [golang image](https://hub.docker.com/_/golang/), with the commands
+```
+docker build -f build/docker/Dockerfile -t basicwebapp-image .
+docker run -itd --rm --name basicwebapp \
+  -p 8080:8080 \
+  --env MESSAGE_SERVICE=mock \
+  basicwebapp-image
+```
+
+This will run the container with the mock messaging service. We will describe
+how to configure the app to connect to Cloud SQL below.
+
+Check that the application is running OK use the same curl commands as above.
+
+Stop the app with the command
+```
+docker stop basicwebapp
+```
+
+Build the Docker container with cloud build:
+```
+gcloud builds submit --config build/cb-docker.yaml .
+```
+
+Run in Docker, test that it works, and stop it
+```
+gcloud auth configure-docker
+PROJECT_ID=[your project]
+docker run -itd --rm -p 8080:8080 \
+  --name basicwebapp \
+  --env MESSAGE_SERVICE=mock \
+  gcr.io/$PROJECT_ID/basicwebapp
+docker ps
+curl -I http://localhost:8080
+docker stop basicwebapp
+```
+
+## Deploy the app to Flex
+To deploy the app to Flex with the database dependency mocked out use the
+command
+```
+gcloud app deploy app-mock.yaml
+```
+
+Check that the app is being served by App Engine with the command
+```
+gcloud app browse
+```
+
+To run the app together with the Cloud SQL database follow steps as in [Using 
+Cloud SQL for 
+MySQL](https://cloud.google.com/appengine/docs/flexible/go/using-cloud-sql) for
+Flex. You can do this by editing the app.yaml file, filling in the text for the
+password. Then redeploy the app using the command
+```
+gcloud app deploy .
+```
+
+You will want to avoid leaving the project id and password in the app
+configuration file, especially if you are adding the file to a code repository.
+Revert the values back to their original symbols by editing app.yaml. To deploy
+with Cloud Build, set the strings as environment variables using the command
+```
+DB_PASSWORD=[user db password]
+gcloud builds submit \
+  --substitutions=_DB_PASSWORD=$DB_PASSWORD \
+  --config build/cb-deploy.yaml .
+```
+
+The [build/cb-deploy.yaml](build/cb-deploy.yaml) file contains a [custom build
```

Comment:
Looks like the code fence above messed up formatting for the rest of the file - this is showing up italicized and code blocks are reversed.

Reply:
I am not sure how to fix it, it looks ok in my editor.